### PR TITLE
doc: update the steps of customize the default StorageClass

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
related issue: https://github.com/harvester/harvester/issues/4045

### CODE REVIEW

File: docs/rancher/csi-driver.md [hunk 1]
```diff
@@ -143,6 +143,34 @@ The only difference is in generating the `cloud-init` config where you need to s
 ./generate_addon_csi.sh <serviceaccount name> <namespace> k3s
 ```
 
+## Customize the default StorageClass
```
Comment (jillian-maroket) [inline docs/rancher/csi-driver.md:146]:
```suggestion
## Customize the Default StorageClass
```

File: docs/rancher/csi-driver.md [hunk 2]
```diff
@@ -143,6 +143,34 @@ The only difference is in generating the `cloud-init` config where you need to s
...
+
+Harvester CSI Driver provides the interface to pre-define the default StorageClass. If we do not specify the default StorageClass, the default StorageClass of Harvester CSI Driver will be the same as Harvester cluster.
+
+To customize the default StorageClass, you can use the parameter `host-storage-class`. Please check with the following example:
```
Comment (jillian-maroket) [inline docs/rancher/csi-driver.md:150]:
```suggestion
The Harvester CSI driver provides the interface for defining the default StorageClass. If the default StorageClass in unspecified, the Harvester CSI driver and the Harvester cluster use the same default StorageClass. 

You can use the parameter `host-storage-class` to customize the default StorageClass.
```

File: docs/rancher/csi-driver.md [hunk 3]
```diff
@@ -143,6 +143,34 @@ The only difference is in generating the `cloud-init` config where you need to s
...
+
+1. Create new a StorageClass of the host Harvester cluster
+
+  ![](/img/v1.3/rancher/create-new-sc.png)
```
Comment (jillian-maroket) [inline docs/rancher/csi-driver.md:154]:
```suggestion
1. Create a StorageClass for the host Harvester cluster.

    Example:
    ![](/img/v1.3/rancher/create-new-sc.png)
```

File: docs/rancher/csi-driver.md [hunk 4]
```diff
@@ -143,6 +143,34 @@ The only difference is in generating the `cloud-init` config where you need to s
...
+
+2. Deploy CSI driver with parameter `host-storage-class`
+
+  ![](/img/v1.3/rancher/deploy-csi-driver-with-host-storage-class.png)
```
Comment (jillian-maroket) [inline docs/rancher/csi-driver.md:158]:
```suggestion
1. Deploy the CSI driver with the parameter `host-storage-class`.

    Example:
    ![](/img/v1.3/rancher/deploy-csi-driver-with-host-storage-class.png)
```

File: docs/rancher/csi-driver.md [hunk 5]
```diff
@@ -143,6 +143,34 @@ The only difference is in generating the `cloud-init` config where you need to s
...
+
+Then, make sure the Harvester CSI driver is ready. You can verify with the following steps.
```
Comment (jillian-maroket) [inline docs/rancher/csi-driver.md:160]:
```suggestion
1. Verify that the Harvester CSI driver is ready.
```

Comment (jillian-maroket) [reply docs/rancher/csi-driver.md:160]: I converted this to a step. All other steps that follow are substeps of this one.

File: docs/rancher/csi-driver.md [hunk 6]
```diff
@@ -143,6 +143,34 @@ The only difference is in generating the `cloud-init` config where you need to s
...
+
+1. Create volume with harvester CSI driver
+
+  ![](/img/v1.3/rancher/create-volume-with-harvester-csi-driver.png)
```
Comment (jillian-maroket) [inline docs/rancher/csi-driver.md:164]:
```suggestion
    1. On the **PersistentVolumeClaims** screen, create a PVC. Select **Use a Storage Class to provision a new Persistent Volume** and specify the StorageClass you created.
        
        Example:
        ![](/img/v1.3/rancher/create-volume-with-harvester-csi-driver.png)
```

File: docs/rancher/csi-driver.md [hunk 7]
```diff
@@ -143,6 +143,34 @@ The only difference is in generating the `cloud-init` config where you need to s
...
+
+2. Check the volume status `bound` and its PVC name
+  
+    ![](/img/v1.3/rancher/check-volume-and-pvc-name.png)
```
Comment (jillian-maroket) [inline docs/rancher/csi-driver.md:168]:
```suggestion
    2. Once the PVC is created, note the name of the provisioned volume and verify that the status is **Bound**.

        Example:
        ![](/img/v1.3/rancher/check-volume-and-pvc-name.png)
```

File: docs/rancher/csi-driver.md [hunk 8]
```diff
@@ -143,6 +143,34 @@ The only difference is in generating the `cloud-init` config where you need to s
...
+
+3. Check the PVC name on the host harvester volume page. Then you can see it is created by the StorageClass you configured.
+
+  ![](/img/v1.3/rancher/check-pvc-name-on-host-harvester-volume-page.png)
```
Comment (jillian-maroket) [inline docs/rancher/csi-driver.md:172]:
```suggestion
    3. On the **Volumes** screen, verify that the volume was provisioned using the StorageClass that you created.
    
        Example:
        ![](/img/v1.3/rancher/check-pvc-name-on-host-harvester-volume-page.png)
```

File: docs/rancher/csi-driver.md (hunk 6 above)
Comment (jillian-maroket) [reply docs/rancher/csi-driver.md:164]: I changed the description to match what we expect users to do on the UI. In this case, they are supposed to create a PVC, not a volume. Also, there is no mention of the CSI driver on this screen.

File: docs/rancher/csi-driver.md (hunk 8 above)
Comment (jillian-maroket) [reply docs/rancher/csi-driver.md:172]: I changed the description to match what we expect users to do on the UI. In this case, they must match the volume name and StorageClass name.

File: docs/rancher/csi-driver.md (hunk 2 above)
Comment (jillian-maroket) [reply docs/rancher/csi-driver.md]: > Harvester CSI driver will use the default StorageClass of the host Harvester cluster.

Thanks for clarifying. The relationship was not very clear to me. We can change sentence #2 to this: 
"If the default StorageClass in unspecified, the Harvester CSI driver uses the default StorageClass of the host Harvester cluster."

**Review by jillian-maroket — APPROVED** [review]

File: docs/rancher/csi-driver.md (hunk 2 above)
Comment (Vicente-Cheng) [reply docs/rancher/csi-driver.md]: updated! Thanks!

**Comment by Vicente-Cheng** [comment]
After the review is complete, I will copy this improvement to v1.1 and v1.2.

**Review by bk201 — APPROVED** [review]

**Pull request merged**